lib/jellyfish/component/hls/request_handler.ex

```diff
@@ -162,13 +188,24 @@ defmodule Jellyfish.Component.HLS.RequestHandler do
     end
   end
 
-  defp update_status(status, last_partial) do
+  defp wait_for_partial_ready(room_id, filename) do
+    GenServer.cast(registry_id(room_id), {:preload_hint, room_id, filename, self()})
+
+    receive do
+      :preload_hint_ready ->
+        EtsHelper.get_partial(room_id, filename)
+    end
+  end
```

Comment:
`wait_for_manifest_ready` and `wait_for_partial_ready`: two functions with similar names, yet totally different APIs. I'd say we should either rename one of them, or make them behave similarly

Reply:
I don't see the problem those are private functions. The names describe really well what they do.